test/testcafe/spec/NumberChallengePushView_spec.js

```diff
@@ -0,0 +1,104 @@
+  .respond(numberChallenge)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/poll')
+  .respond(success);
+
+const numberChallengeWaitMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(numberChallenge)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/poll')
+  .respond(numberChallenge)
+  .onRequestTo('http://localhost:3000/idp/idx/resend')
+  .respond(numberChallenge);
+
+const serverErrorMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(numberChallenge)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/poll')
+  .respond(serverError);
+
+
+fixture('Number Challenge Okta Verify Push');
+
+async function setup(t) {
+  const challengeOktaVerifyPushPageObject = new ChallengeOktaVerifyPushPageObject(t);
+  await challengeOktaVerifyPushPageObject.navigateToPage();
+  return challengeOktaVerifyPushPageObject;
+}
+
+test
+  .requestHooks(numberChallengeSuccessMock)('number challenge screen has right labels', async t => {
+    const challengeOktaVerifyPushPageObject = await setup(t);
+
+    const { log } = await t.getBrowserConsoleMessages();
+    await t.expect(log.length).eql(3);
+    await t.expect(log[0]).eql('===== playground widget ready event received =====');
+    await t.expect(log[1]).eql('===== playground widget afterRender event received =====');
+    await t.expect(JSON.parse(log[2])).eql({
+      controller: 'mfa-verify',
+      formName: 'challenge-poll',
+      authenticatorKey: 'okta_verify',
+      methodType: 'push',
+    });
+    const pageTitle = challengeOktaVerifyPushPageObject.getFormTitle();
+    await t.expect(pageTitle).contains('Push notification sent');
+  });
+
+test
+  .requestHooks(numberChallengeWaitMock)('Warning callout appears after 30 seconds', async t => {
+    const challengeOktaVerifyPushPageObject = await setup(t);
+    await t.wait(30500);
+    const warningBox = challengeOktaVerifyPushPageObject.getWarningBox();
+    await t.expect(warningBox.innerText)
+      .eql('Haven\'t received a push notification yet? Try opening the Okta Verify app on your device or resend the push notification.');
+  });
+
+test
+  .requestHooks(logger, numberChallengeWaitMock)('Calls resend when we click the resend link from within the warning modal', async t => {
+    const challengeOktaVerifyPushPageObject = await setup(t);
+    await t.wait(30500);
+    let warningBox = challengeOktaVerifyPushPageObject.getWarningBox();
+    await t.expect(warningBox.visible).ok();
+    await t.expect(warningBox.innerText)
+      .eql('Haven\'t received a push notification yet? Try opening the Okta Verify app on your device or resend the push notification.');
+    // click resend number challenge
+    challengeOktaVerifyPushPageObject.clickResendNumberChallenge();
+    await t.expect(logger.count(() => true)).eql(8);
+    const req = logger.requests[7].request;
+    const reqBody = JSON.parse(req.body);
+    await t.expect(reqBody).eql({
```

Comment:
these tests have some parts in common. Since both add 30s wait we can get rid of first one ?